### PR TITLE
refactor(CI): remove large Ubuntu runners

### PR DESCRIPTION
**Pull request by iuioiua**
This change is an experiment and alternative to #3230. It removes the use of `xl` Ubuntu runners, as there doesn't appear to be anything computationally heavy in CI that'd justify having them. Since `std/node` was recently removed, this might be more accurate than ever.

### CODE REVIEW

**Comment by iuioiua** [comment]
I'm not sure what's broken CI.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -45,23 +45,21 @@ jobs:
 
       - name: Upload coverage
         uses: codecov/codecov-action@v3
-        if: |
-          !startsWith(matrix.os, 'ubuntu')
+        if: matrix.os != ubuntu-22.04
```
Comment (kt3k) [inline .github/workflows/ci.yml:48]: Looks like a syntax error in actions' yaml. You can see the error at the bottom of this page https://github.com/denoland/deno_std/actions/runs/4349140842

```suggestion
        if: matrix.os != 'ubuntu-22.04'
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -45,23 +45,21 @@ jobs:
...
         with:
           name: ${{ matrix.os }}-${{ matrix.deno }}
           files: cov.lcov
 
       - name: Remove coverage report
         shell: bash
-        if: |
-          !startsWith(matrix.os, 'ubuntu')
+        if: matrix.os != ubuntu-22.04
```
Comment (kt3k) [inline .github/workflows/ci.yml:55]:
```suggestion
        if: matrix.os != 'ubuntu-22.04'
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -45,23 +45,21 @@ jobs:
...
         run: |
           rm -rf ./cov/
           rm cov.lcov
 
       - name: Release if version change
         if: |
-          startsWith(matrix.os, 'ubuntu') &&
+          matrix.os = ubuntu-22.04 &&
```
Comment (kt3k) [inline .github/workflows/ci.yml:62]:
```suggestion
          matrix.os == 'ubuntu-22.04' &&
```

**Comment by iuioiua** [comment]
Wow, that was it? 🤣

**Review by kt3k — APPROVED** [review]
LGTM. Thanks, Asher!

**Pull request merged**